### PR TITLE
Use List/Set injection into configuration

### PR DESCRIPTION
**Pull request by wendigo**
This makes configuration more explicit in regard to expected types. Mostly a cleanup

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by wendigo** [comment]
Configuration classes should get expected types, rather than parsing them. This aligns better with the idea of introduction of alternate formats to Java properties (like TOML)

File: service/trino-verifier/src/main/java/io/trino/verifier/VerifierConfig.java [hunk 1]
```diff
@@ -184,19 +180,9 @@ public Set<QueryType> getControlQueryTypes()
...
+    public VerifierConfig setControlQueryTypes(Set<QueryType> controlQueryTypes)
     {
-        if (Strings.isNullOrEmpty(types)) {
-            this.controlQueryTypes = ImmutableSet.of();
-            return this;
-        }
-
-        ImmutableSet.Builder<QueryType> builder = ImmutableSet.builder();
-        for (String value : Splitter.on(',').trimResults().omitEmptyStrings().split(types)) {
-            builder.add(QueryType.valueOf(value.toUpperCase(ENGLISH)));
```
Comment (losipiuk) [inline service/trino-verifier/src/main/java/io/trino/verifier/VerifierConfig.java:196]: did you loose the logic of uppercasing?

File: service/trino-verifier/src/main/java/io/trino/verifier/VerifierConfig.java [hunk 2]
```diff
@@ -207,19 +193,9 @@ public Set<QueryType> getTestQueryTypes()
...
+    public VerifierConfig setTestQueryTypes(Set<QueryType> testQueryTypes)
     {
-        if (Strings.isNullOrEmpty(types)) {
-            this.testQueryTypes = ImmutableSet.of();
-            return this;
-        }
-
-        ImmutableSet.Builder<QueryType> builder = ImmutableSet.builder();
-        for (String value : Splitter.on(',').trimResults().omitEmptyStrings().split(types)) {
-            builder.add(QueryType.valueOf(value.toUpperCase(ENGLISH)));
```
Comment (losipiuk) [inline service/trino-verifier/src/main/java/io/trino/verifier/VerifierConfig.java:219]: did you loose the logic of uppercasing

File: plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/config/OpenLineageListenerConfig.java [hunk 3]
```diff
@@ -79,12 +75,9 @@ public Set<QueryType> getIncludeQueryTypes()
 
     @Config("openlineage-event-listener.trino.include-query-types")
     @ConfigDescription("Which query types emitted by Trino should generate OpenLineage events. Other query types will be filtered out.")
-    public OpenLineageListenerConfig setIncludeQueryTypes(List<String> includeQueryTypes)
+    public OpenLineageListenerConfig setIncludeQueryTypes(Set<QueryType> includeQueryTypes)
     {
-        this.includeQueryTypes = includeQueryTypes.stream()
-                .map(value -> value.toUpperCase(ENGLISH))
```
Comment (losipiuk) [inline plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/config/OpenLineageListenerConfig.java:85]: did you loose the logic of uppercasing

**Review by losipiuk — COMMENTED** [review]
Seems mostly fine - but not 100% backward compatible.

File: service/trino-verifier/src/main/java/io/trino/verifier/VerifierConfig.java (hunk 1 above)
Comment (wendigo) [reply service/trino-verifier/src/main/java/io/trino/verifier/VerifierConfig.java:196]: no. enum is injected directly.

File: service/trino-verifier/src/main/java/io/trino/verifier/VerifierConfig.java (hunk 2 above)
Comment (wendigo) [reply service/trino-verifier/src/main/java/io/trino/verifier/VerifierConfig.java:219]: no. enum is injected directly.

File: plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/config/OpenLineageListenerConfig.java (hunk 3 above)
Comment (wendigo) [reply plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/config/OpenLineageListenerConfig.java:85]: no. enum is injected directly.

File: plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/config/OpenLineageListenerConfig.java [hunk 4]
```diff
@@ -95,13 +88,10 @@ public Set<OpenLineageTrinoFacet> getDisabledFacets()
 
     @Config("openlineage-event-listener.disabled-facets")
     @ConfigDescription("Which facets should be removed from OpenLineage events.")
-    public OpenLineageListenerConfig setDisabledFacets(List<String> disabledFacets)
+    public OpenLineageListenerConfig setDisabledFacets(Set<OpenLineageTrinoFacet> disabledFacets)
             throws RuntimeException
     {
-        this.disabledFacets = disabledFacets.stream()
-                .map(value -> value.toUpperCase(ENGLISH))
```
[the remark this answers is not in the document]
Comment (wendigo) [reply plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/config/OpenLineageListenerConfig.java:102]: no. enum is injected directly.

**Comment by wendigo** [comment]
@losipiuk it is, string.toUpperCase was needed to call Enum.valueOf but this is handled by airlift.

**Comment by wendigo** [comment]
@losipiuk there are tests in both Trino and Airlift that are setting configuration properties mapped to enum with a lower case strings

**Comment by losipiuk** [comment]
ok

**Review by losipiuk — APPROVED** [review]

**Comment by wendigo** [comment]
@losipiuk i.e. here: https://github.com/trinodb/trino/blob/44a827795e50cf1e98c493eb7d214ed657d04477/plugin/trino-openlineage/src/test/java/io/trino/plugin/openlineage/TestOpenLineageListenerConfig.java#L64

**Comment by colebow** [comment]
Does this need a release note? @wendigo

**Comment by wendigo** [comment]
@colebow no.

**Pull request merged**